src/twisted/persisted/aot.py

```diff
@@ -231,13 +231,17 @@ def indentify(s):
     out = []
     stack = []
     l = ["", s]
+
+    def readline():
+        return l.pop().replace("\0", "")
```

Comment:
```suggestion
        # tokenize no longer supports null bytes in source code since 3.12
        # see https://github.com/python/cpython/issues/102856
        return l.pop().replace("\0", "")
```